internal/dkg/execution.go

```diff
@@ -19,6 +19,26 @@ import (
 	"github.com/drand/kyber/sign/schnorr"
 )
 
+func (d *Process) executeDKG(ctx context.Context, beaconID string) error {
+	// set up the DKG broadcaster for first so we're ready to broadcast DKG messages
+	dkgConfig, err := d.setupDKG(ctx, beaconID)
+	if err != nil {
+		return err
+	}
+
+	d.log.Infow("DKG execution setup successful", "beaconID", beaconID)
+
+	go func(config dkg.Config) {
+		// wait for `KickOffGracePeriod` to allow other nodes to set up their broadcasters
+		time.Sleep(d.config.KickoffGracePeriod)
+		err := d.executeAndFinishDKG(ctx, beaconID, config)
+		if err != nil {
+			d.log.Errorw("there was an error during the DKG!", "beaconID", beaconID, "error", err)
+		}
+	}(*dkgConfig)
+	return nil
```

Comment:
shouldn't this have some kind of `ctx` watch system in case it is canceled early?
Same comment about `select` and  `<-time.After ` as above